### PR TITLE
Cleanup callable for marshalling '_translations' property.

### PR DESCRIPTION
**Pull request by ADmad**
Removed unnecessary loop.

Also fixed setting of errors for translated records in the primary entity.
For .e.g. the error should be set for '_translations.{lang}.{field}' not '{lang}.{field}', for the error to be properly displayed in the form.

### CODE REVIEW

File: tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php [hunk 1]
```diff
@@ -1847,7 +1854,7 @@ public function testBuildMarshalMapBuildEntitiesValidationErrors()
                 '_empty' => 'This field cannot be left empty',
             ],
         ];
-        $this->assertEquals($expected, $entity->getError('es'));
+        $this->assertEquals($expected, $entity->getError('_translations.es'));
```
Comment (markstory) [inline tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php:1857]: This seems like the kind of change that could cause problems for userland code as the key errors are stored in is changing. This does match up with the documentation better now, but I suspect we might still get an issue about the change.

Comment (ADmad) [reply tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php:1857]: With the current array structure the errors wouldn't be displayed under the related fields using `FormHelper::control()`.

There's a possibility the someone could be using `FormHelper::error()` to fetch and display the errors at appropriate location, but that seems quite unlikely. 

That said I can move the error key change to 4.next if we want to be extra safe.

Comment (markstory) [reply tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php:1857]: I'm fine either way. We should think about what our response will be if we get issues that the behavior 'broke'.

Comment (ADmad) [reply tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php:1857]: Meh, I will just move the array keying fix to 4.next :)

Comment (ADmad) [reply tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php]: Removed the error key changes and just kept the cleanup.

**Pull request merged**